### ISSUE TITLE
[BUG/ISSUE]GEOS-Chem restarts cannot be read correctly in a CO2 simualation

### ISSUE THREAD

**Issue by lin02060757** [geoschem/geos-chem#1171]
### Description of the problem
I run a CO2 simulation via GEOSCHEM 13.1.0 with GEOSFP metadata.

I can run the simulation smoothly by directly setting the simulation period from 20150103-20150105, but if I stopped the simulation at 20150104, then restarted the simulation with the restart file of 20150104, the continued simulation failed with the restart file reading problem.

I tried to update GEOS-Chem to 13.4.0 or extend the simulation period to 20150201 then restart,  the problem still existed. 

### Description of troubleshooting performed
```console
Initialize TMPU1    from restart file
 Initialize SPHU1    from restart file
 Initialize PS1_WET  from restart file
 Initialize PS1_DRY  from restart file
 Initialize DELP_DRY from restart file
     - Found all I3     met fields for 2015/01/04 03:00
===============================================================================
R E S T A R T   F I L E   I N P U T

Min and Max of each species in restart file [mol/mol]:
 Detailed log:
**Species   1,      CO2: Min = ***************  Max = *************** Sum = ***************** ( cannot be read correctly from restart)
===============================================================================
Min and Max of each species in BC file [mol/mol]:
Species   1,       CO2: Use background = 3.549999965E-04
GET_BOUNDARY_CONDITIONS: Done applying BCs at 2015/01/04 00:00
 **bad sum at:           91           1          39   0.0000000000000000**      
 ### I, J, L     :           91           1          39
 ### BLTOP       :   -9.9158699720391599E+030
 ### IMIX        :            0
 ### P0        :   -9.9999998482432073E+030
 ### PBL_TOP_hPa :   -9.9158699720391599E+030
 ### SCALE_HEIGHT :    7600.0000000000000     
 ### PL        :    28.367809999999999     
(......)
===============================================================================
GEOS-Chem ERROR: Error in computing F_OF_PBL !
COMPUTE_PBL_HEIGHT ("pbl_mix_mod.F90")
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "COMPUTE_PBL_HEIGHT" at initialization!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...
```

### GEOS-Chem version
GEOS-Chem 13.1.0 and 13.3.4

### Log files
[GC.log.150104-150105.txt](https://github.com/geoschem/geos-chem/files/8249914/GC.log.150104-150105.txt)
[GEOSChem.Restart.20150104_0000z.nc4.zip](https://github.com/geoschem/geos-chem/files/8249918/GEOSChem.Restart.20150104_0000z.nc4.zip)
[HISTORY.rc.txt](https://github.com/geoschem/geos-chem/files/8249922/HISTORY.rc.txt)
[input.geos.txt](https://github.com/geoschem/geos-chem/files/8249939/input.geos.txt)

### Software versions
- CMake version: 3.21.1
- Compilers (Intel or GNU, and version): gcc 10.2.0

**Comment by yantosca** [comment]
Thanks for writing @lin02060757.  It looks like your restart file may be corrupted.  Can you open it with a netCDF file viewer such as ncview?  Or if you don't have ncview, you can use the `ncdump  -cts GEOSChem.Restart.20150104.nc` command to view the file contents.  If either of these don't work then it means that your file is corrupted.

**Comment by yantosca** [comment]
Can you also look at the boundary conditions file?  It looks like you may have some negative numbers in there:
```console
Min and Max of each species in BC file [mol/mol]:
Species   1,       CO2: Use background = 3.549999965E-04
```
I would also recommend configuring GEOS-Chem to use the debugging flags.  From your run directory:
```console
cd build
rm CMakeCache.txt
cmake .. -DRUNDIR=.. -DCMAKE_BUILD_TYPE=Debug
make -j
make -j install
cd ..
```
and then rerun the gcclassic executable.  This will flag if you have any out-of-bounds errors or other issues.  

Also, have you made any modifications to the source code, or are you running "out-of-the-box" (i.e. with no modifications).  If you've made modifications then you might want to revert to a clean copy of GEOS-Chem and then see if the error persists.

**Comment by lin02060757** [comment]
> Can you also look at the boundary conditions file? It looks like you may have some negative numbers in there:
> 
> ```
> Min and Max of each species in BC file [mol/mol]:
> Species   1,       CO2: Use background = 3.549999965E-04
> ```
> 
> I would also recommend configuring GEOS-Chem to use the debugging flags. From your run directory:
> 
> ```
> cd build
> rm CMakeCache.txt
> cmake .. -DRUNDIR=.. -DCMAKE_BUILD_TYPE=Debug
> make -j
> make -j install
> cd ..
> ```
> 
> and then rerun the gcclassic executable. This will flag if you have any out-of-bounds errors or other issues.
> 
> Also, have you made any modifications to the source code, or are you running "out-of-the-box" (i.e. with no modifications). If you've made modifications then you might want to revert to a clean copy of GEOS-Chem and then see if the error persists.

I only inserted a few lines to print out the variables regarding to the pbl calculation.

I rerun the simulation with debug version, the error was as follow:

> HEMCO: Opening ./GEOSChem.Restart.20150104_0000z.nc4
     - Found all CN     met fields for 2011/01/01 00:00
     - Found all A1     met fields for 2015/01/04 00:30
     - Found all A3cld  met fields for 2015/01/04 01:30
     - Found all A3dyn  met fields for 2015/01/04 01:30

Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.

Backtrace for this error:
#0  0x7f9fffa433ff in ???
#1  0x7fa00069b750 in ???
#2  0x90435d in __calc_met_mod_MOD_airqnt._omp_fn.1
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/calc_met_mod.F90:422
#3  0x7fa000468051 in GOMP_parallel
	at /tmp/root/spack-stage/spack-stage-gcc-10.2.0-qjdftffa7em3vtz5wbdojjqnux3elvun/spack-src/libgomp/parallel.c:171
#4  0x8fc5c0 in __calc_met_mod_MOD_airqnt
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/calc_met_mod.F90:324
#5  0x55411a in get_met_fields
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/hco_interface_gc_mod.F90:4595
#6  0x597033 in __hco_interface_gc_mod_MOD_hcoi_gc_run
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/hco_interface_gc_mod.F90:1017
#7  0x4b0002 in __emissions_mod_MOD_emissions_run
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/emissions_mod.F90:203
#8  0x4064f6 in geos_chem
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/Interfaces/GCClassic/main.F90:743
#9  0x40bc79 in main
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/Interfaces/GCClassic/main.F90:32

It seems there are something wrong to read the met file or restart file?

And, it seems the background reference of CO2 is positive? and the figure is reasonable
> Min and Max of each species in BC file [mol/mol]:
> Species   1,       CO2: Use background = 3.549999965E-04

**Comment by lin02060757** [comment]
@yantosca 

Hi, Bob, 

I reran a global simulation with 4x5 MERRA2 for the BC file. Then I ran an Asian-scale simulation with 0.25x0.315 GEOS-FP. 
I met an error described as follows:

![image](https://user-images.githubusercontent.com/76841219/159395858-a6610e7a-2471-479f-8640-ed2f17d24800.png)

It seems the simulation can successfully run for the first two hours but crashed at the 3rd hour.

there is the simulation log （with debug version）:
[GC.log.2.txt](https://github.com/geoschem/geos-chem/files/8320539/GC.log.2.txt)

[HISTORY.rc.txt](https://github.com/geoschem/geos-chem/files/8320544/HISTORY.rc.txt)
[input.geos.txt](https://github.com/geoschem/geos-chem/files/8320545/input.geos.txt)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/8320546/HEMCO_Config.rc.txt)

Look forward to your reply!

**Comment by yantosca** [comment]
From the error log it looks like there was a problem reading in the boundary conditions file:
```console
Backtrace for this error:
#0  0x7f4905a783ff in ???
#1  0x5a8763 in __hco_utilities_gc_mod_MOD_get_boundary_conditions
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/hco_utilities_gc_mod.F90:2478
#2  0x5973db in __hco_interface_gc_mod_MOD_hcoi_gc_run
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/hco_interface_gc_mod.F90:1051
#3  0x4b0002 in __emissions_mod_MOD_emissions_run
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/GeosCore/emissions_mod.F90:203
#4  0x407388 in geos_chem
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/Interfaces/GCClassic/main.F90:965
#5  0x40bc79 in main
	at /home/cara/scratch/co2gfp/CodeDir/src/GEOS-Chem/Interfaces/GCClassic/main.F90:32
```
The error happened at line 2478 of hco_utilities_gc_mod.F90, which is in this block of code:

https://github.com/geoschem/geos-chem/blob/9ab6931db5b556eafd0e002e9fd4fdbcbd5fbb44/GeosCore/hco_utilities_gc_mod.F90#L2455-L2493

You can try to uncomment the debug print statements to see if you can get output about which netCDF file is problematic.

Also -- a segmentation fault can mean that your simulation has run out of memory.  Please also make sure that you have sufficient memory for your system to run the 0.25 x 0.3125 CO2 simulation.  You can reduce memory by turning off diagnostics that aren't needed.

**Comment by lin02060757** [comment]
@yantosca 
Hi bob,

(1) I tried to uncomment the debug print statements, but the recording lines wasn't found in the new GC.log (attached).
 
[GC.log.uncomment debug.txt](https://github.com/geoschem/geos-chem/files/8329781/GC.log.uncomment.debug.txt)

>You can try to uncomment the debug print statements to see if you can get output about which netCDF file is problematic.

(2) I also tried to monitor the occupation situation of memory during the simulation, and for the whole running, the occupation was below 25%. 

![image](https://user-images.githubusercontent.com/76841219/159626452-3913e832-d8c9-4ed3-8be8-c8b6f2eaddcd.png)

>Also -- a segmentation fault can mean that your simulation has run out of memory. Please also make sure that you have sufficient memory for your system to run the 0.25 x 0.3125 CO2 simulation. You can reduce memory by turning off diagnostics that aren't needed.

**Comment by lin02060757** [comment]
Hi,

I have resolved the problem of reading both the BC files and the restart files.

For BC files, I originally set up the BC file reading according to the official guide, but actually, the modification should be made to change the hour field from 0-23 to * in the HEMCO_Config.rc for BC reading [https://github.com/geoschem/geos-chem/issues/248](url). And the hourly BC files were required in my simulation rather than the 3-hour ones. 

For restart files reading, I tried the simulation with GCClassic 13.3.4, and it worked.
